packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -38,122 +93,602 @@ import '@spectrum-web-components/popover/sp-popover.js';
+            </div>
+            ${this.renderHelpText(this.invalid)}${this.renderPicker()}
         `;
     }
 
-    public renderPicker(): TemplateResult {
-        const isDisabled = this.disabled || this.readonly;
+    private renderStateIcons(): TemplateResult | typeof nothing {
+        if (this.invalid)
+            return html`
+                <sp-icon-alert id="invalid" class="icon"></sp-icon-alert>
+            `;
+
+        return nothing;
+    }
+
+    public renderPicker(): TemplateResult | typeof nothing {
+        if (this.readonly) return nothing;
 
         return html`
-            <div class="picker">
-                ${this.renderStateIcons()}
-
-                <sp-picker-button
-                    class="picker-button"
-                    ?open=${this.open}
-                    ?quiet=${this.quiet || this.readonly}
-                    ?invalid=${this.invalid}
-                    ?disabled=${isDisabled}
-                    @click=${this.showPicker}
-                >
-                    <slot name="calendar-icon" slot="icon">
-                        <sp-icon-calendar></sp-icon-calendar>
-                    </slot>
-                </sp-picker-button>
-
-                <!-- TODO: Enable "receives-focus" when calendar is navigable via keyboard -->
-                <sp-overlay
-                    placement="bottom-start"
-                    receives-focus="false"
-                    .triggerElement=${this as HTMLElement}
-                    ?open=${this.open}
-                    @sp-closed=${this.hidePicker}
+            <sp-picker-button
+                ?open=${this.isCalendarOpen}
+                ?quiet=${this.quiet}
+                ?invalid=${this.invalid}
+                ?disabled=${this.disabled}
+                @click=${() => (this.isCalendarOpen = true)}
+            >
+                <slot name="calendar-icon" slot="icon">
+                    <sp-icon-calendar></sp-icon-calendar>
+                </slot>
+            </sp-picker-button>
+
+            <sp-overlay
+                .triggerElement=${this.input}
+                placement="top"
+                offset="0"
+                receives-focus="true"
+                ?open=${this.isCalendarOpen}
+                @sp-closed=${() => (this.isCalendarOpen = false)}
+            >
+                <sp-popover>
+                    <div class="popover-content">
+                        <sp-calendar
+                            .value=${this.value}
+                            .min=${this.min}
+                            .max=${this.max}
+                            @change=${this.handleChange}
+                        ></sp-calendar>
+                    </div>
+                </sp-popover>
+            </sp-overlay>
+        `;
+    }
+
+    private handleChange(event: Event): void {
+        event.stopPropagation();
+        this.isCalendarOpen = false;
+        const calendarValue = (event.target as Calendar).value as CalendarDate;
```

Comment:
Feels wrong that a type assertion is needed here. This is needed because the Calendar component allows its value to be set to more than one type but it's internally converted to `CalendarDate`. 

If a consumer uses other types, i.e. ZonedDateTime, and uses the Calendar component, i see 2 options:
1. He converts that value to `CalendarDate` before passing it to the Calendar component
```ts
import {toCalendarDate} from '@internationalized/date';
<sp-calendar
    .value=${toCalendarDate(myZonedDateTime)}
    @change=${(event: Event) => {
          // Now when he gets that value it will always be of type CalendarDate
          console.log((event.target as Calendar).value)
     }}
></sp-calendar>


```
2. As of now, he can dirrectly pass any DateValue type (`CalendarDate | CalendarDateTime | ZonedDateTime`), but a type assertion is needed when reading the value

Personally, now I'm leaning more towards the second option, as it aligns more with the meaning of the Calendar component.